File: mail-mapper/app/enums/email_label.py
from enum import Enum


class EmailLabel(str, Enum):
    APPLIED = "applied"
    MEETING_INV = "meeting_inv"

    MEETING_ACTION = "meeting_action"
    MEETING_CRT = "meeting_crt"
    MEETING_UPD = "meeting_upd"
    MEETING_CNCL = "meeting_cncl"
    MEETING_UNDEF = "meeting_undef"

    OFFER = "offer"
    DENIED = "denied"
    PENDING = "pending"

    @classmethod
    def is_valid(cls, value: str) -> bool:
        return value in cls._value2member_map_

    @classmethod
    def get_all_end_labels(cls) -> list[str]:
        return [
            EmailLabel.APPLIED,
            EmailLabel.MEETING_INV,
            EmailLabel.MEETING_ACTION,
            EmailLabel.OFFER,
            EmailLabel.DENIED,
            EmailLabel.PENDING,
        ]

File: google-sheets-accessor/internal/pkg/enums/emaillabel.go
package enums

type EmailLabel string

const (
	EmailLabelApplied     EmailLabel = "applied"
	EmailLabelMeetingInv  EmailLabel = "meeting_inv"
	EmailLabelMeetingCrt  EmailLabel = "meeting_crt"
	EmailLabelMeetingUpd  EmailLabel = "meeting_upd"
	EmailLabelMeetingCncl EmailLabel = "meeting_cncl"
	EmailLabelOffer       EmailLabel = "offer"
	EmailLabelDenied      EmailLabel = "denied"
	EmailLabelPending     EmailLabel = "pending"
)

var emailLabelValues = []EmailLabel{
	EmailLabelApplied,
	EmailLabelMeetingInv,
	EmailLabelMeetingCrt,
	EmailLabelMeetingUpd,
	EmailLabelMeetingCncl,
	EmailLabelOffer,
	EmailLabelDenied,
	EmailLabelPending,
}

func (l EmailLabel) IsValid() bool {
	for _, label := range emailLabelValues {
		if l == label {
			return true
		}
	}

	return false
}

func EmailLabelAll() []EmailLabel {
	result := make([]EmailLabel, len(emailLabelValues))
	copy(result, emailLabelValues)

	return result
}
